src/olympia/blocklist/utils.py

```diff
@@ -36,18 +38,23 @@ def block_activity_log_save(
         'comments': f'{len(changed_versions)} versions added to block; '
         f'{len(blocked_versions)} total versions now blocked.',
     }
+    version_details = {}
     if submission_obj:
         details['signoff_state'] = submission_obj.SIGNOFF_STATES.get(
             submission_obj.signoff_state
         )
         if submission_obj.signoff_by:
             details['signoff_by'] = submission_obj.signoff_by.id
+        details['soft'] = version_details['soft'] = (
+            submission_obj.block_type == BlockVersion.BLOCK_TYPE_CHOICES.SOFT_BLOCKED
+        )
```

Comment:
imo we need to change the content of `details` significantly - it's useful to know the `soft`/`hard` state of the submission, but `blocked_versions` and `added_versions` don't mean what they used to mean, in the world of soft blocks.

My suggestion would be more like
'hard_blocked_versions' : ...
'soft_blocked_versions': ...
'added_versions' ...
'changed_versions' ...

Reply:
I'm not sure how useful those keys are (I know `changed_versions` is used in the history we show) ... Do you think this is enough:

```diff
diff --git a/src/olympia/blocklist/utils.py b/src/olympia/blocklist/utils.py
index 896e5274f1..1502f58a2d 100644
--- a/src/olympia/blocklist/utils.py
+++ b/src/olympia/blocklist/utils.py
@@ -10,6 +10,32 @@ from olympia.versions.models import Version
 log = olympia.core.logger.getLogger('z.amo.blocklist')
 
 
+def blocked_versions_activity_details(obj, *, deleted=False):
+    hard_blocked_versions = (
+        []
+        if deleted
+        else sorted(
+            ver.version
+            for ver in obj.addon_versions
+            if ver.is_blocked and not ver.blockversion.soft
+        )
+    )
+    soft_blocked_versions = (
+        []
+        if deleted
+        else sorted(
+            ver.version
+            for ver in obj.addon_versions
+            if ver.is_blocked and ver.blockversion.soft
+        )
+    )
+
+    return {
+        'hard_blocked_versions': hard_blocked_versions,
+        'soft_blocked_versions': soft_blocked_versions,
+    }
+
+
 def block_activity_log_save(
     obj,
     change,
@@ -19,24 +45,25 @@ def block_activity_log_save(
 
     action = amo.LOG.BLOCKLIST_BLOCK_EDITED if change else amo.LOG.BLOCKLIST_BLOCK_ADDED
     addon_versions = {ver.id: ver.version for ver in obj.addon_versions}
-    blocked_versions = sorted(
-        ver.version for ver in obj.addon_versions if ver.is_blocked
-    )
+    blocked_versions_details = blocked_versions_activity_details(obj)
     changed_version_ids = (
         [v_id for v_id in submission_obj.changed_version_ids if v_id in addon_versions]
         if submission_obj
         else sorted(ver.id for ver in obj.addon_versions if ver.is_blocked)
     )
     changed_versions = sorted(addon_versions[ver_id] for ver_id in changed_version_ids)
+    total_versions_blocked = len(
+        blocked_versions_details['hard_blocked_versions']
+    ) + len(blocked_versions_details['soft_blocked_versions'])
 
     details = {
         'guid': obj.guid,
-        'blocked_versions': blocked_versions,
         'added_versions': changed_versions,
         'url': obj.url,
         'reason': obj.reason,
         'comments': f'{len(changed_versions)} versions added to block; '
-        f'{len(blocked_versions)} total versions now blocked.',
+        f'{total_versions_blocked} total versions now blocked.',
+        **blocked_versions_details,
     }
     version_details = {}
     if submission_obj:
@@ -72,11 +99,7 @@ def block_activity_log_save(
 def block_activity_log_delete(obj, deleted, *, submission_obj=None, delete_user=None):
     assert submission_obj or delete_user
     addon_versions = {ver.id: ver.version for ver in obj.addon_versions}
-    blocked_versions = (
-        sorted(ver.version for ver in obj.addon_versions if ver.is_blocked)
-        if not deleted
-        else []
-    )
+    blocked_versions_details = blocked_versions_activity_details(obj, deleted=deleted)
     changed_version_ids = (
         [v_id for v_id in submission_obj.changed_version_ids if v_id in addon_versions]
         if submission_obj
@@ -87,15 +110,18 @@ def block_activity_log_delete(obj, deleted, *, submission_obj=None, delete_user=
         for ver_id in changed_version_ids
         if ver_id in addon_versions
     )
+    total_versions_blocked = len(
+        blocked_versions_details['hard_blocked_versions']
+    ) + len(blocked_versions_details['soft_blocked_versions'])
 
     details = {
         'guid': obj.guid,
-        'blocked_versions': blocked_versions,
         'removed_versions': changed_versions,
         'url': obj.url,
         'reason': obj.reason,
         'comments': f'{len(changed_versions)} versions removed from block; '
-        f'{len(blocked_versions)} total versions now blocked.',
+        f'{total_versions_blocked} total versions now blocked.',
+        **blocked_versions_details,
     }
     action = (
         amo.LOG.BLOCKLIST_BLOCK_EDITED

```

Reply:
The main use case is how the logs are used, yeah.... - in logs.html it looks like we just expose `added_versions` and `removed_versions` - so beyond that there probably isn't much use in adding anything, unless we want to change what's exposed in logs.html too...

Reply:
https://github.com/mozilla/addons-server/pull/22765/commits/ea79c09deb87c725761bdda3db0f7c82bf516e78